llvm/lib/IR/Verifier.cpp

```diff
@@ -5008,6 +5008,9 @@ void Verifier::visitProfMetadata(Instruction &I, MDNode *MD) {
       Check(mdconst::dyn_extract<ConstantInt>(MDO),
             "!prof brunch_weights operand is not a const int");
     }
+  } else {
+    Check(ProfName == "VP", "expected either branch_weights or VP profile name",
```

Comment:
As a followup, it would be nice to verify the structure of VP data, and that it is applied to valid instructions only.

Reply:
oh, good point!